### ISSUE TITLE
building v0.9.1 with py27 and mingw64 got unresolved external symbol _gfortran_st_write referenced in function d1mach_ 

### ISSUE THREAD

**Issue by allenling** [scipy/scipy#14792]
### Describe your issue.

for some reasons, i need to build scipy 0.9.1 + numpy v1.13.3 + openblas v0.3.5 + mingw64 + py27

now i can build numpy + openblas + mingw64 successfully

and when i built scipy with numpy, i got this error 

```
mach.lib(d1mach.o) : error LNK2019: unresolved external symbol _gfortran_st_write referenced in function d1mach_
mach.lib(d1mach.o) : error LNK2019: unresolved external symbol _gfortran_st_write_done referenced in function d1mach_
mach.lib(d1mach.o) : error LNK2019: unresolved external symbol _gfortran_stop_numeric referenced in function d1mach_
mach.lib(d1mach.o) : error LNK2019: unresolved external symbol _gfortran_transfer_character_write referenced in function d1mach_
mach.lib(d1mach.o) : error LNK2019: unresolved external symbol _gfortran_transfer_integer_write referenced in function d1mach_
mach.lib(d1mach.o) : error LNK2019: unresolved external symbol _gfortran_stop_string referenced in function d1mach_
```

when i checked the link cmd, it seemed that no gofrtran lib name provided, here is link cmd

```
error: Command "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\BIN\amd64\link.exe /DLL /nologo /INCREMENTAL:NO /LIBPATH:D:\autodesk\Maya2018\Python\libs /LIBPATH:D:\msys64\mingw64\lib\gcc\x86_64-w64-mingw32\10.3.0 /LIBPATH:D:\autodesk\Maya2018\Python\libs /LIBPATH:D:\autodesk\Maya2018\Python\PC\VC6 /LIBPATH:build\temp.win-amd64-2.7 /LIBPATH:D:\autodesk\Maya2018\Python\libs /LIBPATH:D:\autodesk\Maya2018\Python\PC\VC6 /LIBPATH:build\temp.win-amd64-2.7 quadpack.lib mach.lib openblas.lib openblas.lib /EXPORT:init_quadpack build\temp.win-amd64-2.7\Release\scipy\integrate\_quadpackmodule.obj /OUT:build\lib.win-amd64-2.7\scipy\integrate\_quadpack.pyd /IMPLIB:build\temp.win-amd64-2.7\Release\scipy\integrate\_quadpack.lib" failed with exit status 1120
```

here is my fortran path

```
D:\scipy-0.19.1>gfortran
gfortran: fatal error: no input files
compilation terminated.

D:\scipy-0.19.1>where gfortran
D:\msys64\mingw64\bin\gfortran.exe
D:\Strawberry\c\bin\gfortran.exe
```

### Reproducing Code Example

```python
setup.py build -v
```

### Error message

```shell
error: Command "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\BIN\amd64\link.exe /DLL /nologo /INCREMENTAL:NO /LIBPATH:D:\autodesk\Maya2018\Python\libs /LIBPATH:D:\msys64\mingw64\lib\gcc\x86_64-w64-mingw32\10.3.0 /LIBPATH:D:\autodesk\Maya2018\Python\libs /LIBPATH:D:\autodesk\Maya2018\Python\PC\VC6 /LIBPATH:build\temp.win-amd64-2.7 /LIBPATH:D:\autodesk\Maya2018\Python\libs /LIBPATH:D:\autodesk\Maya2018\Python\PC\VC6 /LIBPATH:build\temp.win-amd64-2.7 quadpack.lib mach.lib openblas.lib openblas.lib /EXPORT:init_quadpack build\temp.win-amd64-2.7\Release\scipy\integrate\_quadpackmodule.obj /OUT:build\lib.win-amd64-2.7\scipy\integrate\_quadpack.pyd /IMPLIB:build\temp.win-amd64-2.7\Release\scipy\integrate\_quadpack.lib" failed with exit status 1120
```
```


### SciPy/NumPy/Python version information

>>> numpy.__version__,sys.version_info
 ('1.13.3', sys.version_info(major=2, minor=7, micro=11, releaselevel='final', serial=0))

**Comment by ilayn** [comment]
It might happen that your gfortran or something in your tool chain is too new.

**Comment by allenling** [comment]
em....

i use gfortran 10.3.0, maybe i should try gcc 5.x

```
D:\scipy-0.19.1>gfortran --version
GNU Fortran (Rev5, Built by MSYS2 project) 10.3.0
Copyright (C) 2020 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

**Comment by carlkl** [comment]
For py-2.7 you can still try the good old mingwpy approach: https://anaconda.org/carlkl/mingwpy

**Comment by carlkl** [comment]
Three years ago I created windows wheels for numpy-1.14.4 as well as scipy-1.1.0 targeting Python-27 based on libmsvcr120. I can summarize the processes that led to this result at that time. This Could take some time, but somewhere I have all patches and scripts needed for this approach.

BTW: The latest numpy and scipy versions available for Python-2.7 are: numpy-1.16.6 and scipy-1.2.2. Is there a reason to use an even older scipy version? The simplest approach is to simply use the windows wheels available at Pypi.

**Comment by allenling** [comment]
thank you @carlkl 

we hit a MKL DLL loading error when we ran numpy that installed from pypi in Maya env, yes we do use a non-standard python interpreter called mayapy.
so our purpose is to strip off MKL from numpy, and we want to use OpenBLAS only.

why compile numpy-scipy and OpenBLAS manually?
because we are not sure that there is any difference between a standard python 2.7 interpreter and the maya python.
what we found was that the nuance is some C interfaces, but do not sure the impact
maybe the best way is compiling all of them manually.

about the version numbers, why numpy v1.13.3 and scipy v0.19.1?
because they are the versions we are running
but they do not have to be 1.13.3 and 0.19.1 we think, maybe any ones work in py2.7 env is ok
we will test them in our maya env
